Source/path.cpp

```diff
@@ -56,6 +57,36 @@ const Displacement PathDirs[8] = {
  */
 int8_t path_directions[9] = { 5, 1, 6, 2, 0, 3, 8, 4, 7 };
 
+bool IsTileNotSolid(Point position)
+{
+	return !nSolidTable[dPiece[position.x][position.y]];
+}
+
+bool IsTileSolid(Point position)
+{
+	if (position.x < 0 || position.y < 0 || position.x >= MAXDUNX || position.y >= MAXDUNY) {

```

Comment:
This is the proper form for an `InDungeonBounds` check. Could we extract this code to a function (and rename the missiles.cpp version until the logic issues in MissileProcs are fixed 😂 )?

This test is used in multiple places in the code as well. I haven't flagged any of them for this PR, mentioning it more to say this would be useful in more places than just these two functions.

Reply:
Totally agree, but decided not to do so for fear I might apply it to all cases and grow the PR further